src/ep_worker.c

```diff
@@ -9848,18 +9854,39 @@ static ep_stat_t w_form_query_getall_insert_new(worker_data_t *wd, obj_info_t *o
         strcat_safe(query, "], ", query_size);
     }
     /* The last index is calculated in the object's table. */
-    if (idx_params_num > 1)
-        strcat_safe(query, "IFNULL(MAX(t0.", query_size);
+     if (BackendIndex == -1 || subst_other_obj || idx_params_num > 1)
+    {
+
+        strcat_safe(query, "IFNULL('", query_size);
+        strcat_safe(query, newrow->be_keys[BackendIndex], query_size);
+        if (idx_params_num > 1)
+            strcat_safe(query, "', MAX(t0.", query_size);
+        else
+            strcat_safe(query, "', MAX(t1.", query_size);
+
+        strcat_safe(query, "[", query_size);
+        strcat_safe(query, idx_params[idx_params_num-1], query_size);
+        strcat_safe(query, "])+1) ", query_size);
+    }
     else
-        strcat_safe(query, "IFNULL(MAX(t1.", query_size);
+    {
+        if (idx_params_num > 1)
+            strcat_safe(query, "IFNULL(MAX(t0.", query_size);
+        else
+            strcat_safe(query, "IFNULL(MAX(t1.", query_size);
 
-    strcat_safe(query, "[", query_size);
-    strcat_safe(query, idx_params[idx_params_num-1], query_size);
-    strcat_safe(query, "]),0)+1 ", query_size);
+        strcat_safe(query, "[", query_size);
+        strcat_safe(query, idx_params[idx_params_num-1], query_size);
+        strcat_safe(query, "]),0)+1 ", query_size);
```

Comment:
Is `+1` needed for case when we use inplicit `XIndex` field, for example, `BSSIndex` and return it from getall backend script?

Reply:
The whole logic if following:

When we add keys to the table we made it in the following manner (e.g. for Device.A.{i}.B.{i}.C.{i}., and getall call with Device.A.{i}.KeyA, Device.A.{i}.B.{i}.KeyB, KeyC answer):
```
INSERT INTO TableC_ValuesTbl 
  (AIndex, BIndex, CIndex, KeyC)
  SELECT (A.AIndex, B.BIndex, (MAX(C.CIndex)+1), KeyCValueFromGetallAnswer)
  FROM
    TableC_ValuesTbl as C
    LEFT JOIN TableB_ValuesTbl as B ON (C.BIndex = B.BIndex)
    LEFT JOIN TableA_ValuesTbl as A ON (C.AIndex = A.AIndex)
  WHERE
    B.KeyB = KeyBValueFromGetallAnswer
    AND A.KeyA = KeyAValueFromGetallAnswer
```

Currently we in situation when `KeyC` is actually `CIndex`. Idea of the changes is to use `KeyCValueFromGetallAnswer` for `CIndex` instead of `(MAX(C.CIndex)+1)` in select request.